jsonutil/puritan/puritan.go

```diff
@@ -0,0 +1,329 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package puritan
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"unicode"
+	"unicode/utf16"
+	"unicode/utf8"
+	"unsafe"
+
+	"github.com/snapcore/snapd/strutil"
+)
+
+// SimpleStrings are JSON strings that are representable with no backslashes
+// (any backslash will be rejected, even if it represents an ASCII character).
+type SimpleString struct {
+	s string
+}
+
+func NewSimpleString(s string) SimpleString {
+	return SimpleString{s}
+}
+
+func (un *SimpleString) UnmarshalJSON(in []byte) error {
+	s, err := unmarshal(in, uOpt{basic: true})
+	if err != nil {
+		return err
+	}
+	un.s = s
+	return nil
+}
+
+// Clean returns the string.
+func (un SimpleString) Clean() string {
```

Comment:
```
// Clean returns the string with all non-ASCII characters removed
```

Reply:
as discussed in-person :-D, saying this would be misleading; I've added a comment to UnmarshalJSON to reiterate things.